rc/extra/dart.kak

```diff
@@ -0,0 +1,106 @@
+# https://dartlang.org/
+#
+
+# Detection
+# ‾‾‾‾‾‾‾‾‾
+
+hook global BufCreate .*\.dart %{
+    set-option buffer filetype dart
+}
+
+# Highlighters
+# ‾‾‾‾‾‾‾‾‾‾‾‾
+
+add-highlighter shared/dart regions
+add-highlighter shared/dart/code default-region group
+add-highlighter shared/dart/back_string region '`' '`' fill string
+add-highlighter shared/dart/double_string region '"' (?<!\\)(\\\\)*" fill string
+add-highlighter shared/dart/single_string region "'" (?<!\\)(\\\\)*' fill string
+add-highlighter shared/dart/comment region /\* \*/ fill comment
+add-highlighter shared/dart/comment_line region '//' $ fill comment
+
+add-highlighter shared/dart/code/ regex %{-?([0-9]*\.(?!0[xX]))?\b([0-9]+|0[xX][0-9a-fA-F]+)\.?([eE][+-]?[0-9]+)?i?\b} 0:value
+
+evaluate-commands %sh{
+    # Grammar
+    keywords="abstract|do|import|super|as|in|switch|assert|else|interface|sync\*|async"
+    keywords="${keywords}|enum|is|this|async\*|export|library|throw|await|external|mixin|break|extends"
+    keywords="${keywords}|new|try|case|factory|typedef|catch|operator|class|final|part"
+    keywords="${keywords}|const|finally|rethrow|while|continue|for|return|with|covariant"
+    keywords="${keywords}|get|set|yield|default|if|static|yield\*|deferred|implements"
+
+    types="void|bool|num|int|double|dynamic|var"
+    values="false|true|null"
+
+    annotations="@[a-zA-Z]+"
+    functions="(_?[a-z][a-zA-Z0-9]*)(\(|\w+=>)"
+    classes="[A-Z][a-zA-Z0-9]*"
+
+    # Add the language's grammar to the static completion list
+    printf %s\\n "hook global WinSetOption filetype=dart %{
+        set-option window static_words ${keywords} ${attributes} ${types} ${values}
+    }" | tr '|' ' '
+
+    # Highlight keywords
+    printf %s "
+        add-highlighter shared/dart/code/ regex \b(${keywords})\b 0:keyword
+        add-highlighter shared/dart/code/ regex \b(${attributes})\b 0:attribute
+        add-highlighter shared/dart/code/ regex \b(${types})\b 0:type
+        add-highlighter shared/dart/code/ regex \b(${values})\b 0:value
+        add-highlighter shared/dart/code/ regex \b(${functions}) 2:function
+        add-highlighter shared/dart/code/ regex \b(${annotations})\b 0:meta
+        add-highlighter shared/dart/code/ regex \b(${classes})\b 0:module
```

Comment:
Some symbols that start/end with a punctuation sign won't be highlighted because of the `\b` escape sequence in the regex.

Example: `\bsync\*\b`, the second `\b` sequence will trigger a match-miss.

Reply:
Great catch! I've amended my commit.

Reply:
Actually you'll have the same issue with annotations (start with `@`), and functions that start with `_` or end with `(`/`>`.